File: Year II/First Semester/MAP/Laborator/Laborator2/decorator/DelayTaskRunner.java
package sem1_2.decorator;

import java.time.Duration;

public class DelayTaskRunner extends AbstractTaskRunner{

    public DelayTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask() {
        try{
            Thread.sleep(Duration.ofSeconds(3));
            super.executeOneTask();
        }
        catch(InterruptedException e){
            System.out.println(e.getMessage());
        }
    }
}

File: Year II/First Semester/MAP/Laborator/Laborator9/ConsoleApp1/ConsoleApp1/Decorator/DelayTaskRunner.cs
namespace ConsoleApp1.Decorator;

public class DelayTaskRunner: AbstractTaskRunner
{
    public DelayTaskRunner(TaskRunner taskRunner) : base(taskRunner) { }

    public override void ExecuteOneTask()
    {
        TaskRunner.ExecuteOneTask();
        try
        {
            Thread.Sleep(3000);
        }
        catch (Exception exception)
        {
            Console.WriteLine(exception.Message);
        }
    }
}